projects/igniteui-angular/src/lib/drop-down/drop-down.component.ts

```diff
@@ -452,10 +452,12 @@ export class IgxDropDownBase implements OnInit, IToggleView {
 
     public selectItem(item?) {
         if (!item) {
```

Comment:
`item` should not be an optional parameter on an public API method. Make it mandatory, remove the checks inside, and do the necessary changes where the method is called.